commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java

```diff
@@ -30,37 +28,24 @@
 
 public class AesEncryptionProvider implements Encryption, Decryption {
 
-    public static final int KEY_SIZE = 512;
-    public static final int KEY_AND_AAD_SIZE_BYTES = KEY_SIZE / 8 / 2;
+    public static final int KEY_SIZE = 256;
     public static final String AES_TRANSFORMATION = "AES/GCM/NoPadding";
 
     private final KeyGenerator aesKeyGenerator;
 
-    static KeyGenerator keyGenerator() {
+    public AesEncryptionProvider() {
         try {
-            final KeyGenerator kg = KeyGenerator.getInstance("AES", "BC");
-            kg.init(KEY_SIZE, SecureRandom.getInstanceStrong());
-            return kg;
+            this.aesKeyGenerator = KeyGenerator.getInstance("AES", "BC");
+            this.aesKeyGenerator.init(KEY_SIZE, SecureRandom.getInstanceStrong());
         } catch (final NoSuchAlgorithmException | NoSuchProviderException e) {
             throw new RuntimeException(e);
         }
     }
 
-    public AesEncryptionProvider() {
-        this.aesKeyGenerator = keyGenerator();
-    }
-
-    public SecretKey createDataKey() {
-        return aesKeyGenerator.generateKey();
-    }
-
     public DataKeyAndAAD createDataKeyAndAAD() {
-        final byte[] dataKeyAndAAD = createDataKey().getEncoded();
-        final byte[] dataKey = new byte[KEY_AND_AAD_SIZE_BYTES];
-        System.arraycopy(dataKeyAndAAD, 0, dataKey, 0, 32);
-        final byte[] aad = new byte[KEY_AND_AAD_SIZE_BYTES];
-        System.arraycopy(dataKeyAndAAD, 32, aad, 0, 32);
-        return new DataKeyAndAAD(new SecretKeySpec(dataKey, "AES"), aad);
+        final var dataKey = aesKeyGenerator.generateKey();
+        final byte[] aad = aesKeyGenerator.generateKey().getEncoded();
```

Comment:
Should reuse the data key?
```suggestion
        final var dataKey = aesKeyGenerator.generateKey();
        final byte[] aad = dataKey.getEncoded();
```

Reply:
No, these two are separate things, they must be different

Reply:
Made a note and test for this